pkg/apis/core/v1alpha1/types_controllerinstallation.go

```diff
@@ -60,6 +60,7 @@ type ControllerInstallationStatus struct {
 	// +optional
 	Conditions []Condition `json:"conditions,omitempty"`
 	// ProviderStatus contains type-specific status.
+	// DEPRECATED: This field will be removed in a future version.
```

Comment:
Why do you want to remove this? We have to keep it as we don't know who installs the extension. Gardener implements a standard installer for `helm`-typed registrations, but there might be other types, and here the responsible installer might want to store status in the `providerStatus` field. Please revert this or explain why you think it's no longer needed.